packages/frontend/app/components/bulk-new-users.js

```diff
@@ -234,18 +236,12 @@ export default class BulkNewUsersComponent extends Component {
 
   parseFile = restartableTask(async (file) => {
     const proposedUsers = await this.getFileContents(file);
-    const existingUsernames = await this.existingUsernames();
-    const filledOutUsers = proposedUsers.map((obj) => {
-      obj.existingUsernames = existingUsernames;
-
-      return obj;
-    });
-    this.validUsers = await filter(filledOutUsers, async (obj) => {
-      return await obj.isValid();
+    this.validUsers = await filter(proposedUsers, async (obj) => {
+      return obj.isValid();
```

Comment:
`ProposedUser.isValid()` is async, so we should await for promise resolution here.

```suggestion
      return await obj.isValid();
```

Reply:
I'm always on the fence about this. Technically we're returning a promise into the `filter` and that filter will await the promise for us and act on the resolved value. So adding the await is unnecessary, but it does make it clearer what is happening.